zulip_bots/zulip_bots/bots/googletranslate/googletranslate.py

```diff
@@ -0,0 +1,103 @@
+# To use this plugin, you need to set up the Google Cloud API key for this bot in
+# googletranslate.conf in this (zulip_bots/bots/googletranslate/) directory.
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+import requests
+
+class GoogleTranslateHandler(object):
+    '''
+    This bot will translate any messages sent to it using google translate.
+    Before using it, make sure you set up google api keys, and enable google
+    cloud translate from the google cloud console.
+    '''
+    def usage(self):
+        return '''
+            This plugin allows users translate messages
+            Users should @-mention the bot with the format
+            @-mention "<text_to_translate>" <target-language> <source-language(optional)>
+            '''
+
+    def initialize(self, bot_handler):
+        self.config_info = bot_handler.get_config_info('googletranslate')
+        self.supported_languages = get_supported_languages(self.config_info['key'])
+
+    def handle_message(self, message, bot_handler):
+        bot_response = get_translate_bot_response(message['content'],
+                                                  self.config_info,
+                                                  message['sender_full_name'],
+                                                  self.supported_languages)
+        bot_handler.send_reply(message, bot_response)
+
+api_url = 'https://translation.googleapis.com/language/translate/v2'
+
+help_text = '''
+Google translate bot
+Please format your message like:
+`@-mention "<text_to_translate>" <target-language> <source-language(optional)>`
+To see a list of languages, @-mention this bot with `languages`
+'''
+
+languages_text_to_translate = '''Please see [here](https://cloud.google.com/translate/docs/languages) for all languages and their codes.
+You can use these codes, or their names.'''
+
+def get_supported_languages(key):
+    parameters = {'key': key, 'target': 'en'}
+    r = requests.get(api_url + '/languages', params = parameters)
```

Comment:
While `r` is an obvious abbreviation if you know what the function does, perhaps choose a more descriptive name for those who initially might not. Good names also indicate to those reading the code later what it is intended to do, which can be useful not just for understanding, but checking that the intent matches what is actually happening :)